backend/test/integration/test_appointment.py

```diff
@@ -124,3 +129,92 @@ def test_dont_show_other_subscribers_appointments(
         assert len(data) == 1
         assert other_appointment.id != data[0]['id']
         assert appointment.id == data[0]['id']
+
+
+class TestCancelAppointment:
+    def test_cancel_appointment(self, with_client, make_appointment, with_db):
+        appointment = make_appointment()
+        payload = {'reason': 'Test cancellation'}
+
+        with patch('appointment.routes.api.Tools') as mock_tools:
+            response = with_client.post(f'/apmt/{appointment.id}/cancel', headers=auth_headers, json=payload)
+
+            assert response.status_code == 200
+            assert mock_tools().send_cancel_vevent.call_count == len(appointment.slots)
+
+        db = with_db()
```

Comment:
You can do `with with_db() as db:` to avoid having to call .close(). I do recognize that's maybe one with too many though 😅